docs/elasticsearch.md

```diff
@@ -1,3 +1,120 @@
+# `elasticsearch` Metrics Receiver
+
+The `elasticsearch` metrics receiver fetches node and cluster level stats from Elasticsearch nodes. The receiver is meant to be run 
+
+## Prerequisites
+Elasticsearch supports collecting metrics out of the box. If [Elasticsearch security features](https://www.elastic.co/guide/en/elasticsearch/reference/7.16/secure-cluster.html) are enabled, a user with the `monitor` or `manage` [cluster privilege](https://www.elastic.co/guide/en/elasticsearch/reference/7.16/security-privileges.html#privileges-list-cluster) must be configured.
+
+## Configurations
+To configure a receiver for your Elasticsearch JSON logs, specify the following fields:
+
+| Field                  | Required | Default                 | Description                                                                                                                                                             |
+|------------------------|----------|-------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `type`                 | required |                         | Must be `elasticsearch`.                                                                                                                                                |
+| `collection_interval`  | optional | `60s`                   | A [time.Duration](https://pkg.go.dev/time#ParseDuration) value, such as `30s` or `5m`.                                                                                  |
+| `endpoint`             | optional | `http://localhost:9200` | Base URL for the Elasticsearch REST API                                                                                                                                 |
+| `username`             | optional |                         | Username for authentication with Elasticsearch. Required if `password` is set.                                                                                          |
+| `password`             | optional |                         | Password for authentication with Elasticsearch. Required if `username` is set.                                                                                          |
+| `collect_jvm_metrics`  | optional | `true`                  | If true, supported JVM metrics will be collected                                                                                                                        |
+| `skip_cluster_metrics` | optional | `true`                  | If true, cluster level metrics will not be collected. To prevent duplicate metrics, this should be set to false for one node only.                                      |
```

Comment:
TBD whether `skip_cluster_metrics` should default to true or false. I'm slightly leaning towards false. Will check with PM offline.

Reply:
Cool, just to give the rationale for default true - If you're installing this on, say, a 20 node cluster, 19 out of 20 of those are gonna want this to be true.

It might make sense to have this default false if it's just really common to have a 1 or 2 node cluster (which I'm not sure whether it is or not.)

Reply:
It's weird to me that we have `collect_jvm_metrics` (a positive) and `skip_cluster_metrics` (a negative). Seems like both should be positive or both negative. I think that's probably true regardless of what the default is.

I'll leave this comment on the scoping doc, but it occurs to me that for Elasticsearch the cluster metrics are basically just count metrics - maybe we can just omit them and calculate them from the other metrics (e.g. `nodes` is just `REDUCE_COUNT` on one of the other node metrics).

Reply:
+1 for changing `skip_cluster_metrics` to `collect_cluster_metrics` for consistency. To summarize the tradeoff of the default value of `collect_cluster_metrics`.

* If `collect_cluster_metrics` default to true
  * Pros: Easier discovery for users to be aware of this metrics
  * Cons: For a large cluster with N nodes, user only need to customize the config to be false on N - 1 node
* If `collect_cluster_metrics` default to false
  * Pros: For a large cluster with many nodes, user only need to customize the config to be true on 1 node
  * Cons: Users might miss cluster metrics unless they read the documentation thoroughly.

Will sync internally with PM on Thursday for this and the clustered application discussion.

Reply:
Collecting only on one node still has the SPOF problem (if that node goes down, you lose your cluster metrics), so maybe we want `true` and live with the duplicate metrics.

Reply:
If we release one version of Ops Agent with `collect_jvm_metrics` default to `false`, I guess it's considered a breaking change if we later want to switch the default to `true`, right?

I'm currently leaning towards keeping it true by default given all the tradeoffs (See more details in the `Duplicate / Overlapping Metrics Issue from Clustered Applications` doc).

Reply:
To clarify - 

`collect_jvm_metrics` should be default true
`collect_cluster_metrics` should be default true

Correct?

Reply:
Correct.